### PR TITLE
Remove `HelperEnv`

### PR DESCRIPTION
**Pull request by gabrielfu**
## Related Issues/PRs

              Is it possible to remove `HelperEnv` in a follow-up PR?

_Originally posted by @harupy in https://github.com/mlflow/mlflow/pull/8888#discussion_r1246080469_
            

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/tracking/fluent/test_fluent.py [hunk 1]
```diff
@@ -169,13 +147,13 @@ def create_experiment(
...
 @pytest.fixture(autouse=True)
-def reset_experiment_id():
+def reset_experiment_id(monkeypatch):
     """
     This fixture resets the active experiment id *after* the execution of the test case in which
     its included
     """
     yield
-    HelperEnv.set_values()
+    monkeypatch.delenvs((MLFLOW_EXPERIMENT_ID.name, MLFLOW_EXPERIMENT_NAME.name), raising=False)
```
Comment (harupy) [inline tests/tracking/fluent/test_fluent.py:156]: We can remove this line. `monkeypatch` restore the environment variables after each test finishes.

Comment (gabrielfu) [reply tests/tracking/fluent/test_fluent.py:156]: you're right

File: tests/tracking/fluent/test_fluent.py [hunk 2]
```diff
@@ -169,13 +147,12 @@ def create_experiment(
 
 
 @pytest.fixture(autouse=True)
-def reset_experiment_id():
+def reset_experiment_id(monkeypatch):
```
Comment (harupy) [inline tests/tracking/fluent/test_fluent.py:150]:
```suggestion
def reset_experiment_id():
```

File: tests/tracking/fluent/test_fluent.py [hunk 3]
```diff
@@ -204,38 +181,39 @@ def _is_function_or_class(obj):
...
     assert _get_experiment_id_from_env() == exp_id
 
     # create experiment from env name
     name = "random experiment %d" % random.randint(1, 1e6)
-    HelperEnv.set_values(name=name)
-    HelperEnv.assert_values(None, name)
+    monkeypatch.delenv(MLFLOW_EXPERIMENT_ID.name, raising=False)
+    monkeypatch.setenv(MLFLOW_EXPERIMENT_NAME.name, name)
     assert MlflowClient().get_experiment_by_name(name) is None
     assert _get_experiment_id_from_env() is not None
```
Comment (harupy) [inline tests/tracking/fluent/test_fluent.py:211]: This block looks strange. The comment says create experiment, but it doesn't.

Comment (gabrielfu) [reply tests/tracking/fluent/test_fluent.py:211]: by setting an experiment name that doesn't exist yet, `_get_experiment_id_from_env()` will create a new experiment of that name

Comment (harupy) [reply tests/tracking/fluent/test_fluent.py:211]: Makes sense!

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**